pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+        const executors = range(4).map(randomAddress);
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, executors, { from: getSender() });
+        const execution = await authorizer.getScheduledExecution(id);
+
+        expect(execution.protected).to.be.true;
+        await Promise.all(
+          executors.map(async (executor) => expect(await authorizer.isExecutor(id, executor)).to.be.true)
+        );
+      });
+
+      it('granter can cancel the execution', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+        expect(await authorizer.isCanceler(id, getSender())).to.be.true;
+
+        const receipt = await authorizer.cancel(id, { from: getSender() });
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionCancelled', { scheduledExecutionId: id });
+      });
+
+      it('can be executed after the expected delay', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        const receipt = await authorizer.execute(id);
+        expectEvent.inReceipt(await receipt.wait(), 'ExecutionExecuted', { scheduledExecutionId: id });
+      });
+
+      it('grants the permission when executed', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.hasPermission(ACTION_1, user, WHERE_1)).to.be.equal(true);
+      });
+
+      it('does not grant any other permissions when executed', async () => {
+        const id = await authorizer.scheduleGrantPermission(ACTION_1, user, WHERE_1, [], { from: getSender() });
+
+        await advanceTime(delay);
+        await authorizer.execute(id);
+
+        expect(await authorizer.hasPermission(ACTION_3, user, WHERE_1)).to.be.equal(false);
+        expect(await authorizer.hasPermission(ACTION_2, user, WHERE_2)).to.be.equal(false);
+      });
+
+      it('emits an event', async () => {
+        const receipt = await authorizer.instance
+          .connect(getSender())
+          .scheduleGrantPermission(ACTION_1, user.address, WHERE_1, []);
+
+        expectEvent.inReceipt(await receipt.wait(), 'GrantPermissionScheduled', {
+          actionId: ACTION_1,
+          account: user.address,
+          where: WHERE_1,
+        });
+      });
+    }
+
+    context('when the sender is root', () => {
+      itScheduleGrantPermissionCorrectly(() => root);
+    });
+
+    context('when the sender is granter', () => {
+      sharedBeforeEach('makes a granter', async () => {
+        await authorizer.addGranter(ACTION_1, granter, EVERYWHERE, { from: root });
+        await authorizer.addGranter(ACTION_2, granter, EVERYWHERE, { from: root });
+      });
+      itScheduleGrantPermissionCorrectly(() => granter);
```

Comment:
Similar to my `grantPermission` suggestion, I'd try to improve handling of scenarios where the action fails because the caller does not have the right permissions.